ui/src/ducks/config.js

```diff
@@ -100,48 +122,50 @@ export function fetchConfigAction() {
   return { type: FETCH_CONFIG };
 }
 
-export function setApiConfigAction(conf) {
+export function setApiConfigAction(conf: Config) {
   return { type: SET_API_CONFIG, payload: conf };
 }
 
 export function setInitialLanguageAction() {
   return { type: SET_INITIAL_LANGUAGE };
 }
 
-export function updateLanguageAction(language) {
+// Todo : this actually seems to be never used and duplicate of setLanguageAction
+export function updateLanguageAction(language: string) {
   return { type: UPDATE_LANGUAGE, payload: language };
 }
 
-export function setUserManagerConfigAction(payload) {
+export function setUserManagerConfigAction(payload: {authority: string,  redirect_uri: string}) {
   return { type: SET_USER_MANAGER_CONFIG, payload };
 }
 
-export function setUserManagerAction(conf) {
+export function setUserManagerAction(conf: UserManager) {
   return { type: SET_USER_MANAGER, payload: conf };
 }
 
-export function setUserLoadedAction(isLoaded) {
+export function setUserLoadedAction(isLoaded: boolean) {
   return { type: SET_USER_LOADED, payload: isLoaded };
 }
 
-export function updateAPIConfigAction(payload) {
+// Todo : this actually seems to be never used and duplicate of setApiConfigAction
+export function updateAPIConfigAction(payload: Config) {
   return { type: UPDATE_API_CONFIG, payload };
 }
 
 export function logoutAction() {
   return { type: LOGOUT };
 }
 
-export function setThemesAction(themes) {
+export function setThemesAction(themes: Themes) {
   return { type: SET_THEMES, payload: themes };
 }
 
 // Selectors
-export const languageSelector = (state) => state.config.language;
-export const apiConfigSelector = (state) => state.config.api;
+export const languageSelector = (state: RootState) => state.config.language;
```

Comment:
Just a question: since `RootState` is properly typed, is flow able to infer the return type of these selectors?

Reply:
Yes, flow is inferring the type of `languageSelector` from `state.config.language`